src/sklearn_compat/_sklearn_compat.py

```diff
@@ -445,6 +445,71 @@ def check_array(
             **kwargs,
         )
 
+    def check_X_y(
+        X,
+        y,
+        accept_sparse=False,
+        *,
+        accept_large_sparse=True,
+        dtype="numeric",
+        order=None,
+        copy=False,
+        force_writeable=False,
+        force_all_finite="deprecated",
+        ensure_all_finite=None,
+        ensure_2d=True,
+        allow_nd=False,
+        multi_output=False,
+        ensure_min_samples=1,
+        ensure_min_features=1,
+        y_numeric=False,
+        estimator=None,
+    ):
+        """Input validation for standard estimators.
+
+        Check the original documentation for more details:
+        https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html
+        """
+        from sklearn.utils.validation import check_X_y as _check_X_y
+
+        if ensure_all_finite is not None and force_all_finite != "deprecated":
+            raise ValueError(
+                "'force_all_finite' and 'ensure_all_finite' cannot be used together. "
+                "Pass `ensure_all_finite` only."
+            )
+        elif ensure_all_finite is None and force_all_finite == "deprecated":
+            force_all_finite = True
+        else:
+            force_all_finite = (
+                ensure_all_finite if ensure_all_finite is not None else force_all_finite
+            )
+
+        check_X_y_params = inspect.signature(_check_X_y).parameters
+        kwargs = (
+            {"force_writeable": force_writeable}
+            if "force_writeable" in check_X_y_params
+            else {}
+        )
```

Comment:
Actually it takes one line less ;) and we can add later new parameters.
```suggestion
        kwargs = {}
        if "force_writeable" in check_array_params:
            kwargs["force_writeable"] = force_writeable
```